src/libs/actions/connections/index.ts

```diff
@@ -363,12 +364,18 @@ function updateManyPolicyConnectionConfigs<TConnectionName extends ConnectionNam
     API.write(WRITE_COMMANDS.UPDATE_MANY_POLICY_CONNECTION_CONFIGS, parameters, {optimisticData, failureData, successData});
 }
 
-function hasSynchronizationError(policy: OnyxEntry<Policy>, connectionName: PolicyConnectionName, isSyncInProgress: boolean): boolean {
+function getSynchronizationErrorMessage(policy: OnyxEntry<Policy>, connectionName: PolicyConnectionName, isSyncInProgress: boolean): string | undefined {
     // NetSuite does not use the conventional lastSync object, so we need to check for lastErrorSyncDate
     if (connectionName === CONST.POLICY.CONNECTIONS.NAME.NETSUITE) {
-        return !isSyncInProgress && !!policy?.connections?.[CONST.POLICY.CONNECTIONS.NAME.NETSUITE].lastErrorSyncDate;
+        if (!isSyncInProgress && !!policy?.connections?.[CONST.POLICY.CONNECTIONS.NAME.NETSUITE].lastErrorSyncDate) {
+            return Localize.translateLocal('workspace.accounting.syncError', connectionName);
+        }
+        return;
+    }
+    const connection = policy?.connections?.[connectionName];
+    if (!isSyncInProgress && connection?.lastSync?.isSuccessful === false) {
+        return connection?.lastSync?.errorMessage;
     }
```

Comment:
Can we do this instead? I think we  should always return something for this function.

```suggestion
      if (isSyncInProgress || !!connection?.lastSync?.isSuccessful) {
        return;
    }
    return connection?.lastSync?.errorMessage;
```

Reply:
Isn't the behaviour the same as now? If the 

```javascript
!isSyncInProgress && connection?.lastSync?.isSuccessful === false
```

is not true the function ends and returns nothing -> undefined